### ISSUE TITLE
Refs Must Have Owner Warning

### ISSUE THREAD

**Issue by und0ck3d** [YouCanBookMe/react-datetime#331]
[X] Bug report
[ ] Feature request
[ ] Support request

### Steps to Reproduce
- Install the latest `react-datetime` version;
- Require and use it in any React component;

### Expected Results
Expected that component worked normally.

### Actual Results
When the previous steps are done, an error shows in the console:

```
Uncaught Error: addComponentAsRefTo(...): Only a ReactOwner can have refs. You might be adding a ref to a component that was not created inside a component's `render` method, or you have multiple copies of React loaded (details: https://fb.me/react-refs-must-have-owner).
```

### Minimal Reproduction of the Problem
1. I'm creating a component to be reused in other projects via installing with NPM. In that component I'm using `react-datetime`.
2. In the main project, I require this new component (mine) and `react-datetime` is provoking the error mentioned before (because if I don't use this package it doesn't trigger the error).

**The following code is redacted for simplicity**

My Component:

```javascript
import React from 'react';
import PropTypes from 'prop-types';
import DateTime from 'react-datetime';

class AvailForm extends React.Component {
  render() {
    let { date } = this.state;

    return (
      <div>
        <DateTime date={ date } onChange={ this._onChangeDate } />
      </div>
    );
  }
}

export default AvailForm;
```

On the main project I do:

```javascript
import React from 'react';
import AvailFrom from 'availform'; // not actual package name, just for eg. purposes

export default class SomePage extends React.Component {
  render() {
    return (
      <AvailForm />
    );
  }
}
```

Finally, on main project's main file I'm rendering the component like (simplified):

```javascript
import { render } from 'react-dom';
import SomePage from './containers/SomePage';

render(<SomePage />, document.getElementById('some-page'));
```

Please, let me know if I haven't been clear in any step or if further information is needed.

Thanks everyone beforehand for your time and help!

**Comment by joaocrq** [comment]
I've been reading around and it seems that the problem was caused because in the main app I had React as a dependency and in my component too, which was causing to having multiple instances of React running. I "fixed" it by setting my component's `react` and `react-dom` dependencies as peerDependencies, however now I can't build the component with Webpack...

I'm not sure if this discussion should continue here, but can anyone help me out please?

**Comment by bruno-gouveia** [comment]
I think the problem comes from react-onclickoutisde dependency, the version 6.1 fixes this.

https://github.com/Pomax/react-onclickoutside

**Comment by exah** [comment]
Yes, thats it. Could someone update dependencies?

**Comment by john-osullivan** [comment]
If updating those dependencies is a quick fix, I'd really appreciate it!  This bug just reared its head in my app, breaking something which was otherwise working fine.

**Comment by john-osullivan** [comment]
As described in that now closed PR, that dependency can't be updated without rewriting `react-datetime` to stop using mixins.  The working version only supports pure class syntax.  @simeg @bruno-gouveia , are you two actively using `react-datetime` and getting around this issue?  Trying to figure out which version I should downgrade to.

**Comment by bruno-gouveia** [comment]
I fixed it updating to version `^6.4.0` and by using `.default` when importing `onClickOutside`. I'll make a pull request with this fix.

**Comment by hoangtrieukd** [comment]
Try it! in webpack.config.js:
externals: {
  'react': 'React'
}

**Comment by layneanderson** [comment]
Fixed via 2.10.3
Thanks to @bruno-gouveia
